src/inmanta/protocol/rest/client.py

```diff
@@ -137,4 +137,11 @@ def _get_client_config(self) -> str:
             LOGGER.exception("Failed to send request")
             return common.Result(code=500, result={"message": str(e)})
 
-        return common.Result(code=response.code, result=self._decode(response.body))
+        if CONTENT_TYPE not in response.headers or response.headers[CONTENT_TYPE] == JSON_CONTENT:
+            return common.Result(code=response.code, result=self._decode(response.body))
+        elif response.headers[CONTENT_TYPE] == HTML_CONTENT:
+            return common.Result(code=response.code, result=response.body.decode("utf-8"))
```

Comment:
perhaps I'm being overprotective, but utf-8 is not the default encoding for text/html  
https://www.w3.org/International/articles/http-charset/index

(it is for json https://www.ietf.org/rfc/rfc4627.txt)